config/default/kustomization.yaml

```diff
@@ -24,49 +24,143 @@ bases:
 # [PROMETHEUS] To enable prometheus monitor, uncomment all sections with 'PROMETHEUS'.
 #- ../prometheus
 
-patchesStrategicMerge:
+patches:
 # Protect the /metrics endpoint by putting it behind auth.
 # If you want your controller-manager to expose the /metrics
 # endpoint w/o any authn/z, please comment the following line.
-- manager_auth_proxy_patch.yaml
-
-
+- path: manager_auth_proxy_patch.yaml
```

Comment:
I expected this patch to change or be deleted, since it's what adds the kube-rbac-proxy sidecar:
 * https://github.com/cert-manager/sample-external-issuer/blob/main/config/default/manager_auth_proxy_patch.yaml

Did you forget to commit something?

Reply:
I could extend the PR to include migration away from `kube-rbac-proxy`, but that will increase the number of changes significantly. But can still make sense. WDYT?

Reply:
Oh, I misunderstood your PR description. 
> But since I noticed some use of deprecated kustomize features, I made some additional improvements.

I thought you meant that you'd removed kube-rbac-proxy **and** additionally replaced some deprecated kustomize features.

Now I understand that this PR is ground work for a follow up PR which removes kube-rbac-proxy.